core/templates/services/contextual/url.service.ts

```diff
@@ -149,15 +149,15 @@ export class UrlService {
     return null;
   }
 
-  getSubtopicUrlFragmentFromLearnerUrl(): string {
+  getSubtopicUrlFragmentFromLearnerUrl(): string | undefined {
```

Comment:
why the `get` method is returning `undefined`?

Reply:
In this function there is a case where it throws an error when the URL is invalid. In that case the typescript checks give me the warning that the function can return undefined.